tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -114,7 +114,7 @@ abstract class AppBase(
           val logFiles = reader.listEventLogFiles
           logFiles.foreach { file =>
             Utils.tryWithResource(openEventLogInternal(file.getPath, fs)) { in =>
-              val lines = Source.fromInputStream(in)(Codec.UTF8).getLines().toList
+              val lines = Source.fromInputStream(in)(Codec.UTF8).getLines().toIterator
```

Comment:
+1, materializing this in Java Heap is unnecessary. toIterator is redundant though because getLines is a line iterator.